### PR TITLE
purl2sym: Add metadata support for packages hosted on GitHub

### PR DESCRIPTION
**Pull request by keshav-space**
- This PR adds support for getting package metadata for arbitrary GitHub repositories using this PURL format: `pkg:github/{owner}/{repo-name}`.
- Also explicitly adds support for these packages hosted on GitHub.

  - [x] pkg:github/avahi/avahi
  - [x] pkg:github/bestouff/genext2fs
  - [x] pkg:github/dosfstools/dosfstools
  - [x] pkg:github/google/brotli
  - [x] pkg:github/hewlettpackard/wireless-tools
  - [x] pkg:github/inotify-tools/inotify-tools
  - [x] pkg:github/libbpf/bpftool
  - [x] pkg:github/llvm/llvm-project
  - [x] pkg:github/nixos/nix
  - [x] pkg:github/plougher/squashfs-tools
  - [x] pkg:github/pupnp/pupnp
  - [x] pkg:github/python/cpython
  - [x] pkg:github/rpm-software-management/rpm
  - [x] pkg:github/shadow-maint/shadow
  - [x] pkg:github/sqlite/sqlite
  - [x] pkg:github/u-boot/u-boot  
  - [x] pkg:generic/miniupnpc
  - [x] pkg:generic/miniupnpd
  - [x] pkg:generic/minissdpd

> [!NOTE]  
> Fetching metadata for arbitrary GitHub repositories assumes that the tag name of the repository is a valid version.

### CODE REVIEW

File: src/fetchcode/package.py [hunk 1]
```diff
@@ -399,11 +376,11 @@ def get_package_info(cls, package_url):
 
         version = package_url.version
         if version and version in IPKG_RELEASES:
-            archive = IPKG_RELEASES[version]
+            archives = IPKG_RELEASES[version]
```
Comment (TG1999) [inline src/fetchcode/package.py:379]: Why is it renamed to 'archives' ? does it contain more than a single archive? And does all archives have a single field "url" and "date"

File: src/fetchcode/utils.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        name = node["name"]
+        target = node["target"]
+
+        # in case the tag is a signed tag, then the commit info is in target['target']
+        if "committedDate" not in target:
+            target = target["target"]
+
+        committed_date = target.get("committedDate")
+        release_date = None
+        if committed_date:
```
Comment (TG1999) [inline src/fetchcode/utils.py:36]: check for parsers error as well, if it's not parsable handle that scenario

File: src/fetchcode/utils.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+def get_response(url):
+    """
+    Generate `Package` object for a `url` string
+    """
+    resp = requests.get(url)
+    if resp.status_code == 200:
+        return resp.json()
+
+    raise Exception(f"Failed to fetch: {url}")
```
Comment (TG1999) [inline src/fetchcode/utils.py:156]: Missing EOL

File: src/fetchcode/package.py (hunk 1 above)
Comment (keshav-space) [reply src/fetchcode/package.py]: Reverted this change is not required.

File: src/fetchcode/utils.py (hunk 2 above)
Comment (keshav-space) [reply src/fetchcode/utils.py:37]: Done

**Review by TG1999 — APPROVED** [review]
LGTM!

**Pull request merged**